core/services/beacon/main.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+import argparse
+import asyncio
+import logging
+import pathlib
+from typing import Dict, List, Optional
+
+import psutil
+from commonwealth.settings.manager import Manager
+from loguru import logger
+from zeroconf import IPVersion
+from zeroconf.asyncio import AsyncServiceInfo, AsyncZeroconf
+
+from settings import Interface, ServiceTypes, SettingsV1
+
+
+class AsyncRunner:
+    def __init__(self, ip_version: IPVersion, interfaces: List[str]) -> None:
+        self.ip_version = ip_version
+        self.aiozc: Optional[AsyncZeroconf] = None
+        self.interfaces: List[str] = interfaces
+        self.services: List[AsyncServiceInfo] = []
+
+    def add_services(self, services: List[AsyncServiceInfo]) -> None:
+        logger.info("Adding services:")
+        for service in services:
+            logger.info(service)
+        self.services.extend(services)
+
+    async def register_services(self) -> None:
+        self.aiozc = AsyncZeroconf(ip_version=self.ip_version, interfaces=self.interfaces)  # type: ignore
+        tasks = [self.aiozc.async_register_service(info, cooperating_responders=True) for info in self.services]
+        background_tasks = await asyncio.gather(*tasks)
+        await asyncio.gather(*background_tasks)
+        logger.info("Finished registration, press Ctrl-C to exit...")
+        while True:
+            await asyncio.sleep(1)
+
+    async def unregister_services(self) -> None:
+        assert self.aiozc is not None
+        tasks = [self.aiozc.async_unregister_service(info) for info in self.services]
+        background_tasks = await asyncio.gather(*tasks)
+        await asyncio.gather(*background_tasks)
+        await self.aiozc.async_close()
+
+
+class Beacon:
+    def __init__(self) -> None:
+        self.runners: List[AsyncRunner] = []
+        self.manager = Manager("Beacon", SettingsV1)
+        self.manager.load()
+        # manager still returns "valid" settings even if file is absent, so we check for the "default" field
```

Comment:
Wait that appears to be an issue, can you open one and add an example to replicate it ?
Can you also tag such issue on this comment with a TODO: ?

Reply:
https://github.com/bluerobotics/BlueOS-docker/issues/880